### PR TITLE
Adding core support when using SPN credentials

### PR DESCRIPTION
**Pull request by 20shivangi**

### CODE REVIEW

File: Tasks/Common/VstsAzureHelpers_/VstsAzureHelpers_.psm1 [hunk 1]
```diff
@@ -11,8 +11,6 @@ if ($global:DebugPreference -eq 'Continue') {
 # Import the loc strings.
 Import-VstsLocStrings -LiteralPath $PSScriptRoot/module.json
 
-[System.Net.WebRequest]::DefaultWebProxy = Get-VstsWebProxy
```
Comment (niadak) [inline Tasks/Common/VstsAzureHelpers_/VstsAzureHelpers_.psm1]: Do we know why it was added? Can this break any exiting scenario?

File: Tasks/Common/VstsAzureHelpers_/Utility.ps1 [hunk 2]
```diff
@@ -5,14 +5,14 @@
         [Switch] $ServicePrincipal
     )
 
-    # Add the certificate to the cert store.
-    $certificate = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2
```
Comment (niadak) [inline Tasks/Common/VstsAzureHelpers_/Utility.ps1]: Is it currently broken in V3 task?If not why this change is required for V4 task?

File: Tasks/SqlAzureDacpacDeploymentV1/task.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "version": {
         "Major": 1,
         "Minor": 156,
-        "Patch": 0
+        "Patch": 1
```
Comment (niadak) [inline Tasks/SqlAzureDacpacDeploymentV1/task.json]: Why this change is required?

File: Tasks/Common/VstsAzureHelpers_/VstsAzureHelpers_.psm1 (hunk 1 above)
Comment (20shivangi) [reply Tasks/Common/VstsAzureHelpers_/VstsAzureHelpers_.psm1]: It is added back.

File: Tasks/SqlAzureDacpacDeploymentV1/task.json (hunk 3 above)
Comment (20shivangi) [reply Tasks/SqlAzureDacpacDeploymentV1/task.json]: Since I made some changes in Common, it was required to bump up version of some of these tasks, so updated patch version

File: Tasks/Common/VstsAzureHelpers_/Utility.ps1 (hunk 2 above)
Comment (20shivangi) [reply Tasks/Common/VstsAzureHelpers_/Utility.ps1]: Even with my change, it is working fine when using powershell.exe
Problem is coming when using pwsh.exe, somehow it is not able to access the import method, so I made this change. 
You can refer this : https://www.cloudnotes.io/x509certificate-is-immutable-on-this-platform-use-the-equivalent-constructor-instead/
